### PR TITLE
create financial/price_plus_tax function

### PR DESCRIPTION
**Pull request by alirezasariri78**
# Pull Request Template

## Description

Please include a summary of the change and which issue (if any) is fixed.
A brief description of the algorithm and your implementation method can be helpful too. If the implemented method/algorithm is not so
well-known, it would be helpful to add a link to an article explaining it with more details.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I ran bellow commands using the latest version of **rust nightly**.
- [x] I ran `cargo clippy --all -- -D warnings` just before my last commit and fixed any issue that was found.
- [x] I ran `cargo fmt` just before my last commit.
- [x] I ran `cargo test` just before my last commit and all tests passed.
- [x] I added my algorithm to the corresponding `mod.rs` file within its own folder, and in any parent folder(s).
- [x] I added my algorithm to `DIRECTORY.md` with the correct link.
- [x] I checked `COUNTRIBUTING.md` and my code follows its guidelines.

Please make sure that if there is a test that takes too long to run ( > 300ms), you `#[ignore]` that or
try to optimize your code or make the test easier to run. We have this rule because we have hundreds of
tests to run; If each one of them took 300ms, we would have to wait for a long time.

### CODE REVIEW

File: src/financial/README.md
Comment (vil02) [inline src/financial/README.md:1]: This file is not needed. Please remove it.

File: src/financial/price_plus_tax.rs
Comment (vil02) [inline src/financial/price_plus_tax.rs:1]: This seems to be _too trivial_. I am not sure if it is needed.

File: src/financial/present_value.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        );
+    }
+
+    #[test]
+    fn test_present_value_zero_discount_rate() {
+        assert_eq!(
+            184924.55,
+            present_value(0.0, vec![109129.39, 30923.23, 15098.93, 29734.0, 39.0]).unwrap()
+        );
+    }
```
Comment (vil02) [inline src/financial/present_value.rs:68]: Could you please express these tests using _macros_, cf. [`shortest_palindrome.rs`](https://github.com/TheAlgorithms/Rust/blob/c906255c51a757d15edbffe2253b8b4225a76a2a/src/string/shortest_palindrome.rs).

File: src/financial/present_value.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        return Err(PresentValueError::EmptyCashFlow);
+    }
+
+    let present_value = cash_flows
+        .iter()
+        .enumerate()
+        .map(|(i, &cash_flow)| cash_flow / (1.0 + discount_rate).powi(i as i32))
+        .sum::<f64>();
+
+    Ok((present_value * 100.0).round() / 100.0)
```
Comment (vil02) [inline src/financial/present_value.rs:21]: Consider creating a separate function doing the rounding. It has to be tested.

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/financial/price_plus_tax.rs
Comment (alirezasariri78) [reply src/financial/price_plus_tax.rs:1]: based on this link in python repository i add this module :
https://github.com/TheAlgorithms/Python/blob/master/financial/price_plus_tax.py

**Comment by alirezasariri78** [comment]
changes are done as you said :)

File: src/financial/present_value.rs (hunk 1 above)
Comment (alirezasariri78) [reply src/financial/present_value.rs]: that is nice idea , ill do it

File: src/financial/present_value.rs
Comment (vil02) [inline src/financial/present_value.rs:1]: Could you please document _here_ what does this file provide?

**Review by vil02 — CHANGES_REQUESTED** [review]

**Comment by vil02** [comment]
@alirezasariri78: please make sure to update your branch - there are some new warnings in clippy and without the update the CI will fail (cf. #838).

**Comment by alirezasariri78** [comment]
document added to module file

**Review by vil02 — APPROVED** [review]

**Pull request merged**